scripts/generate-client.js

```diff
@@ -1,32 +1,30 @@
 'use strict';
+const cp = require('child_process');
 const bluebird = require('bluebird');
 const chalk = require('chalk');
 const config = require('config');
 const path = require('path');
 const fse = bluebird.promisifyAll(require('fs-extra'));
-const exec = bluebird.promisify(require('child_process').exec);
+const exec = bluebird.promisify(cp.exec);
 
-const serverPath = path.join(process.cwd(), 'dist', 'index.js');
-const swaggerSpecPath = path.join(
-    process.cwd(),
-    'dist',
-    'swagger.json'
-);
+const pkgRoot = path.join(__dirname, '../');
+const serverPath = path.join(pkgRoot, 'src/index.js');
+const swaggerSpecPath = path.join(pkgRoot, 'dist', 'swagger.json');
 const server = require(serverPath);
 
 const codegenExecPath = path.join(
-    process.cwd(),
+    pkgRoot,
```

Comment:
you will note that many times throughout this PR there are swaps from `cwd()` to a path string.  assuming `nodeapi/` is root can be a dangerous assumption of the API is run from any other dir